src/libsystemd/sd-login/sd-login.c

```diff
@@ -134,6 +135,220 @@ _public_ int sd_pid_get_cgroup(pid_t pid, char **cgroup) {
         return 0;
 }
 
+static int pidfd_verify_pid(int pidfd, pid_t pid) {
+        pid_t current_pid;
+        int r;
+
+        assert(pidfd >= 0);
+        assert(pid > 0);
+
+        r = pidfd_get_pid(pidfd, &current_pid);
+        if (r < 0)
+                return r;
+
+        return current_pid != pid ? -ESRCH : 0;
+}
```

Comment:
oh, i was thinking that this should exist in process-util.c and that dbus-manager.c would be updated to use it too.